examples/plot_scalable_poly_kernels.py

```diff
@@ -0,0 +1,118 @@
+"""
+=======================================================
+Scalable learning with polynomial kernel aproximation
+=======================================================
+
+This example illustrates the use of :class:`PolynomialSampler` to efficiently
+generate polynomial kernel feature-space approximations. This is used
+to train linear classifiers that approximate the accuracy of kernelized ones.
+
+.. currentmodule:: sklearn.kernel_approximation
+
+We use the Covtype dataset [2]_, trying to reproduce the experiments on the
+original paper of Tensor Sketch [1]_, i.e. the algorithm implemented by
+:class:`PolynomialSampler`.
+
+First, we compute the accuracy of a linear classifier on the original
+features. Then, we train linear classifiers on different numbers of
+features (`n_components`) generated by :class:`PolynomialSampler`,
+approximating the accuracy of a kernelized classifier in a scalable manner.
+"""
+print(__doc__)
+
+# Author: Daniel Lopez-Sanchez <lope@usal.es>
+# License: BSD 3 clause
+from sklearn.datasets import fetch_covtype
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import normalize, MinMaxScaler
+from sklearn.svm import LinearSVC
+from sklearn.kernel_approximation import PolynomialSampler
+from sklearn.pipeline import Pipeline
+
+##############################################################################
+# Load the Covtype dataset, which contains 581,012 samples
+# with 54 features each, distributed among 6 classes. The goal of this dataset
+# is to predict forest cover type from cartographic variables only
+# (no remotely sensed data). After loading, we transform it into a binary
+# classification problem to match the version of the dataset in the
+# LIBSVM webpage [2]_, which was the one used in [1]_.
+
+data = fetch_covtype()
+X, y = data["data"], data["target"]
+
+y[y != 2] = 0
+y[y == 2] = 1  # We will try to separate class 2 from the other 6 classes.
+
+##############################################################################
+# Here we select 20,000 samples for training and 20,000 for testing.
```

Comment:
The example takes 2 minutes to run, I don't know what is the rule for examples in scikit-learn but it seems a bit too long. Since the bottleneck should be the SVC, which scales poorly with the number of samples, we don't need to decrease the number of samples too much to largely decrease the computational time. Would the example still be meaningful with say 5000 or 10000 samples?

Reply:
Reduced the number training samples to `5,000`.